app/controllers/submissions_controller.rb

```diff
@@ -198,6 +236,51 @@ def downloadAll
               filename: "#{@course.name}_#{@course.semester}_#{@assessment.name}_submissions.zip")
   end
 
+  action_auth_level :downloadBatch, :course_assistant
+  def downloadBatch
+    submission_ids = params[:submission_ids]
+    flash[:error] = "Cannot index submissions for nil assessment" if @assessment.nil?
+
+    unless @assessment.valid?
+      @assessment.errors.full_messages.each do |msg|
+        flash[:error] += "<br>#{msg}"
+      end
+      flash[:html_safe] = true
+    end
+
+    if @assessment.disable_handins
+      flash[:error] = "There are no submissions to download."
+      if @cud.course_assistant
+        redirect_to([@course, @assessment])
+      else
+        redirect_to([@course, @assessment, :submissions])
+      end
+      return
+    end
+
+    submissions = submission_ids.map { |sid| @assessment.submissions.find_by(sid) }
```

Comment:
same comment about `.map`, as mentioned in the `destroyBatch` code; there's also a bug right now where you're returning the first submission in the assessment since it should be `find_by(id: sid)`, if you wanted to go this route

Reply:
fixed